### ISSUE TITLE
[Bug]: Type 'SlashCommandOptionsOnlyBuilder' is not assignable to type

### ISSUE THREAD

**Issue by ShkourBashtawi** [ppauel/typescript-discord-bot#19]
### What happened?

When user addStringOption or any options expect addSubComamnd,addSubCommandGroup  it wil return error 

```
Type 'SlashCommandOptionsOnlyBuilder' is not assignable to type 'SlashCommandBuilder | SlashCommandSubcommandsOnlyBuilder | Omit<SlashCommandBuilder, "addSubcommand" | "addSubcommandGroup">'.
  Type 'SlashCommandOptionsOnlyBuilder' is not assignable to type 'Omit<SlashCommandBuilder, "addSubcommand" | "addSubcommandGroup">'.
    The types returned by 'setName(...)' are incompatible between these types.
      Type 'SlashCommandOptionsOnlyBuilder' is missing the following properties from type 'SlashCommandBuilder': addSubcommandGroup, addSubcommand
      ```
      
      this my code
      ```js
 import { PermissionsBitField, SlashCommandBuilder } from 'discord.js';
import { ChatInputCommand } from '../../interfaces';

const command: ChatInputCommand = {
    options: new SlashCommandBuilder()
        .setName('emit')
        .setDescription('Emit an event')
        .setDMPermission(false)
        .setDefaultMemberPermissions(PermissionsBitField.Flags.Administrator)
        .addStringOption((option) =>
            option
                .setName('event')
                .setDescription('The event to emit')
                .setRequired(true),
        )
        .addStringOption((option) =>
            option
                .setName('data')
                .setDescription('The data to emit')
                .setRequired(true),
        ),
    global: true,
    execute: async (_client, interaction) => {
        const event = interaction.options.getString('event', true);
        const data = interaction.options.getString('data', true);
        _client.emit(event, data);
        await interaction.reply({ content: `Emitted event \`${event}\``, ephemeral: true });
    },
};
export default command;
```

### Version

discord.js v14.16.2
nodejs v18.20.4

**Comment by ppauel** [comment]
Hey there @ShkourBashtawi! Thank you for reporting this bug. This issue should be fixed in https://github.com/ppauel/typescript-discord-bot/commit/d9b97855f16bf7e94ba64bf3725aeeb05199336f. I moved from a rest instance to using the client's application class.

Edit: The issue was caused by the Command Interface (https://github.com/ppauel/typescript-discord-bot/commit/61732fcf21fff96b5896b9ac2bd335565bccb2ff).
